packages/crayons-core/src/components/options-list/list-options.tsx

```diff
@@ -40,7 +40,9 @@ export class ListOptions {
       const filteredValue =
         value !== ''
           ? dataSource.filter((option) =>
-              option.text.toLowerCase().includes(value.toLowerCase())
+              option?.[this.optionLabelPath]
```

Comment:
I don't think you need optional chaining for option. u can change it to 
`option[this.optionLabelPath]?.toLowerCase()

Reply:
Please make this changes in other places. option should be a valid object since its passed from `filter` and `map`